pkg/cmd/backuprestore/restoreutils.go

```diff
@@ -0,0 +1,108 @@
+package backuprestore
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var static_pod_list = []string{
+	"etcd-pod.yaml",
+	"kube-apiserver-pod.yaml",
+	"kube-controller-manager-pod.yaml",
+	"kube-scheduler-pod.yaml",
+}
+
+func restore(configDir, dataDir, backupDir string) error {
+
+	var (
+		assetDir           = filepath.Join(configDir, "assets")
+		manifestDir        = filepath.Join(configDir, "manifests")
+		manifestStoppedDir = filepath.Join(assetDir, "manifests-stopped")
+		restoreEtcdPodYaml = filepath.Join(configDir, "static-pod-resources", "etcd-certs", "configmaps", "restore-etcd-pod", "pod.yaml")
+		dataDirBackup      = dataDir + "-backup"
+	)
+	fmt.Printf("configDir=%s,dataDir=%s,backupDir=%s\n", configDir, dataDir, backupDir)
+	// locate snapshot db and static pod resources archive
+	snapshotFile, err := findTheLatestRevision(backupDir, "snapshot_", false)
+	if err != nil {
+		return fmt.Errorf("restore: could not find snapshot db: %w", err)
+	}
+	resourcesArchive, err := findTheLatestRevision(backupDir, "static_kuberesources_", false)
+	if err != nil {
+		return fmt.Errorf("restore: could not find static pod resources: %w", err)
+	}
+
+	// Move manifests to manifests-stopped-dir
+	if err := checkAndCreateDir(manifestStoppedDir); err != nil {
+		return fmt.Errorf("restore: checkAndCreateDir failed: %w", err)
+	}
+	for _, podyaml := range static_pod_list {
+		err := os.Rename(filepath.Join(manifestDir, podyaml), filepath.Join(manifestStoppedDir, podyaml))
+		if err != nil {
+			return fmt.Errorf("restore: attempt to stop %s failed: %w", podyaml, err)
+		}
+	}
+
+	// TODO: Wait for static pods to stop
+	// crictl ps equivalent??
+	time.Sleep(180 * time.Second)
```

Comment:
We'll open a 4.6 blocker to improve the reliability of this check (e.g. through a runtime API call)